submarine-server/server-core/src/test/java/org/apache/submarine/server/rest/ExperimentRestApiTest.java

```diff
@@ -0,0 +1,182 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ExperimentRestApiTest {
+
+  private static ExperimentRestApi experimentRestApi;
+  private static ExperimentManager mockExperimentManager;
+  private final AtomicInteger experimentCounter = new AtomicInteger(0);
+  private Experiment experiment;
+
+  private static GsonBuilder gsonBuilder = new GsonBuilder()
+      .registerTypeAdapter(ExperimentId.class, new ExperimentIdSerializer())
+      .registerTypeAdapter(ExperimentId.class, new ExperimentIdDeserializer());
+  private static Gson gson = gsonBuilder.setDateFormat("yyyy-MM-dd HH:mm:ss").create();
+
+  @BeforeClass
+  public static void init() {
+    mockExperimentManager = mock(ExperimentManager.class);
+    experimentRestApi = new ExperimentRestApi();
+    experimentRestApi.setExperimentManager(mockExperimentManager);
+  }
+
+  @Before
+  public void createAndUpdateExperiment() {
+    experiment = new Experiment();
+    experiment.setAcceptedTime("2020-08-06T08:39:22.000+08:00");
+    experiment.setCreatedTime("2020-08-06T08:39:22.000+08:00");
+    experiment.setRunningTime("2020-08-06T08:39:23.000+08:00");
+    experiment.setFinishedTime("2020-08-06T08:41:07.000+08:00");
+    experiment.setUid("0b617cea-81fa-40b6-bbff-da3e400d2be4");
+    experiment.setName("tf-example");
+    experiment.setStatus("Succeeded");
+    experiment.setExperimentId(ExperimentId.newInstance(SubmarineServer.getServerTimeStamp(),
+        experimentCounter.incrementAndGet()));
+    ExperimentSpec experimentSpec = new ExperimentSpec();
+    EnvironmentSpec environmentSpec = new EnvironmentSpec();
+    environmentSpec.setName("foo");
+    experimentSpec.setEnvironment(environmentSpec);
+    experiment.setSpec(experimentSpec);
+    when(mockExperimentManager.createExperiment(any(ExperimentSpec.class))).thenReturn(experiment);
+    Response createExperimentResponse = experimentRestApi.createExperiment(experimentSpec);
+    assertEquals(Response.Status.OK.getStatusCode(), createExperimentResponse.getStatus());
+    Experiment result = getResultFromResponse(createExperimentResponse, Experiment.class);
+    assertEquals(experiment.getAcceptedTime(), result.getAcceptedTime());
+  }
+
+  @Test
+  public void getExperiment() {
+    when(mockExperimentManager.getExperiment(any(String.class))).thenReturn(experiment);
+    Response getExperimentResponse = experimentRestApi.getExperiment("1");
+    Experiment experiment = getResultFromResponse(getExperimentResponse, Experiment.class);
+    assertEquals("0b617cea-81fa-40b6-bbff-da3e400d2be4", experiment.getUid());
+    assertEquals("2020-08-06T08:39:22.000+08:00", experiment.getAcceptedTime());
+    assertEquals("tf-example", experiment.getName());
+    assertEquals("2020-08-06T08:39:22.000+08:00", experiment.getCreatedTime());
+    assertEquals("2020-08-06T08:39:23.000+08:00", experiment.getRunningTime());
+  }
```

Comment:
We could create a private function to compare `experiment` and `Response`, so that we don't need to write duplicate code in the patch, get, list, ..etc.